cluster-operator/src/main/java/io/strimzi/operator/cluster/operator/assembly/KafkaRebalanceAssemblyOperator.java

```diff
@@ -223,13 +223,32 @@ public void eventReceived(Action action, KafkaRebalance kafkaRebalance) {
                     Reconciliation reconciliation = new Reconciliation("kafkarebalance-watch", kafkaRebalance.getKind(),
                             kafkaRebalance.getMetadata().getNamespace(), kafkaRebalance.getMetadata().getName());
 
-                    LOGGER.debugCr(reconciliation, "EventReceived {} on {} with status [{}] and {}={}", action,
+                    LOGGER.infoCr(reconciliation, "EventReceived {} on {} with status [{}] and {}={}", action,
                             kafkaRebalance.getMetadata().getName(),
                             kafkaRebalance.getStatus() != null ? rebalanceStateConditionType(kafkaRebalance.getStatus()) : null,
                             ANNO_STRIMZI_IO_REBALANCE, rawRebalanceAnnotation(kafkaRebalance));
 
-                    withLock(reconciliation, LOCK_TIMEOUT_MS,
-                        () -> reconcileRebalance(reconciliation, action == Action.DELETED ? null : kafkaRebalance));
+                    if (action == Action.MODIFIED) {
+                        KafkaRebalanceBuilder patchedKafkaRebalance = new KafkaRebalanceBuilder(kafkaRebalance);
+
+                        if (!kafkaRebalance.getMetadata().getAnnotations().containsKey("oldGeneration")) {
+                            patchedKafkaRebalance.editMetadata().addToAnnotations(Map.of("oldGeneration", "1")).endMetadata();
+                        }
+
+                        if (!patchedKafkaRebalance.buildMetadata().getAnnotations().get("oldGeneration").equals(kafkaRebalance.getMetadata().getGeneration().toString())) {
+                            patchedKafkaRebalance.editMetadata()
+                                    .addToAnnotations(Map.of(ANNO_STRIMZI_IO_REBALANCE, KafkaRebalanceAnnotation.refresh.toString()))
+                                    .addToAnnotations(Map.of("oldGeneration", kafkaRebalance.getMetadata().getGeneration().toString()))
+                                    .endMetadata();
```

Comment:
When does the "oldGeneration" annotation value get incremented? Or does it only ever have  one of two values - null or 1?

Reply:
@kyguy it's getting the value from the "generation" assigned to the resource by Kubernetes when the spec changes. Shubham is just saving it in an annotation as a way to know when spec is changed.